lib/components/base_component.dart

```diff
@@ -56,6 +56,7 @@ abstract class BaseComponent extends Component {
   @override
   void update(double dt) {
     _effectsHandler.update(dt);
+    _children.removeWhere((c) => c.shouldRemove).forEach((c) => c.onRemove());
```

Comment:
nice catch! we should probably add a test to this as well (but can be a followup)

Reply:
I added a test for it in #605